src/cbmc/cbmc_solvers.cpp

```diff
@@ -133,23 +161,17 @@ std::unique_ptr<cbmc_solverst::solvert> cbmc_solverst::get_dimacs()
 
 std::unique_ptr<cbmc_solverst::solvert> cbmc_solverst::get_bv_refinement()
 {
-  std::unique_ptr<propt> prop=[this]() -> std::unique_ptr<propt>
-  {
-    // We offer the option to disable the SAT preprocessor
-    if(options.get_bool_option("sat-preprocessor"))
-    {
-      no_beautification();
-      return util_make_unique<satcheckt>();
-    }
-    return util_make_unique<satcheck_no_simplifiert>();
-  }();
-
-  prop->set_message_handler(get_message_handler());
+  const bool use_simplifier=options.get_bool_option("sat-preprocessor");
+  if(use_simplifier)
+    no_beautification();
+  auto prop=make_satcheck(options, use_simplifier, get_message_handler());
 
   bv_refinementt::infot info;
   info.ns=&ns;
   info.prop=prop.get();
   info.ui=ui;
+  std::tie(info.random_seed, info.random_var_freq)=
+    parse_satcheck_options(options);
 
```

Comment:
What the hell is this?

Reply:
I had to Google this too -- it's pattern-matching essentially, like `val (x, y) = f()` in ML or the like. It returns a std::tuple of references, such that if the `parse` function returns `(a, b)` then this code is equivalent to:
```
auto ret=parse_satcheck_options(options);
info.random_seed=ret.first;
info.random_var_freq=ret.second;
```

Reply:
Using C11 best practice (and recent additions to the STL) should be encouraged! @LAJW and @reuk have been very good at calling this out. Yes, it does mean that many (me certainly included) have to learn more about it.